src/lib/logical_query_plan/abstract_lqp_node.hpp

```diff
@@ -301,6 +322,18 @@ class AbstractLQPNode : public std::enable_shared_from_this<AbstractLQPNode>, pr
   virtual std::optional<QualifiedColumnName> _resolve_local_table_name(
       const QualifiedColumnName& qualified_column_name) const;
 
+  static bool _equals(const AbstractLQPNode& lqp_left,
```

Comment:
This is a bunch of `_equals` methods where it takes some time to find the difference. Can we put their purpose into the name?

Reply:
I intentionally didn't do this, because the signature would get even more cluttered and the information would be duplicated in the parameters. I will add comments though.